File: core/src/test/resources/fixtures/complex_nested/expected.java
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class User {
    @SerializedName("user_id")
    public Integer userId;

    @SerializedName("user_name")
    public String userName;

    @SerializedName("is_vip")
    public Boolean isVip;

    @SerializedName("balance")
    public Double balance;

    @SerializedName("tags")
    public List<String> tags;

    @SerializedName("scores")
    public List<List<Integer>> scores;

    @SerializedName("empty_list")
    public List<EmptyList> emptyList;

    @SerializedName("address")
    public Address address;

    @SerializedName("orders")
    public List<Orders> orders;
}

class EmptyList {
}

class Geo {
    @SerializedName("lat")
    public Double lat;

    @SerializedName("lng")
    public Double lng;
}

class Address {
    @SerializedName("city")
    public String city;

    @SerializedName("zip_code")
    public String zipCode;

    @SerializedName("geo")
    public Geo geo;
}

class Items {
    @SerializedName("sku")
    public String sku;

    @SerializedName("qty")
    public Integer qty;
}

class Orders {
    @SerializedName("order_id")
    public String orderId;

    @SerializedName("total")
    public Double total;

    @SerializedName("items")
    public List<Items> items;
}

File: core/src/test/resources/fixtures/complex_nested/expected.ts
export interface User {
  user_id?: number
  user_name?: string
  is_vip?: boolean
  balance?: number
  tags?: string[]
  scores?: number[][]
  empty_list?: EmptyList[]
  address?: Address
  orders?: Orders[]
}

export interface EmptyList {
}

export interface Geo {
  lat?: number
  lng?: number
}

export interface Address {
  city?: string
  zip_code?: string
  geo?: Geo
}

export interface Items {
  sku?: string
  qty?: number
}

export interface Orders {
  order_id?: string
  total?: number
  items?: Items[]
}